services/visual_recognition/v4-collection-utils.js

```diff
@@ -0,0 +1,507 @@
+/**
+ * Copyright 2020 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+module.exports = function(RED) {
+  const SERVICE_IDENTIFIER = 'visual-recognition',
+    VisualRecognitionV4 = require('ibm-watson/visual-recognition/v4'),
+    { IamAuthenticator } = require('ibm-watson/auth'),
+    FEATURE = 'image-feature',
+    REQUIRED_PARAMS = {
+      'analyze': ['collectionIds'],
+      'createCollection': ['name', 'description'],
+      'getCollection': ['collectionId'],
+      'updateCollection': ['collectionId'],
+      'deleteCollection': ['collectionId'],
+      'addImages': ['collectionId'],
+      'listImages': ['collectionId'],
+      'getImageDetails': ['collectionId', 'imageId'],
+      'deleteImage': ['collectionId', 'imageId'],
+      'getJpegImage': ['collectionId', 'imageId'],
+      'train': ['collectionId'],
+      'addImageTrainingData' : ['collectionId', 'imageId', 'objects']
+    };
+
+  var pkg = require('../../package.json'),
+    serviceutils = require('../../utilities/service-utils'),
+    payloadutils = require('../../utilities/payload-utils'),
+    temp = require('temp'),
+    fileType = require('file-type'),
+    fs = require('fs'),
+    fsp = require('fs').promises,
+    sAPIKey = null,
+    apikey = '',
+    service = null,
+    endpoint = '',
+    sEndpoint = '';
+
+  // temp is being used for file streaming to allow the file to arrive so it can be processed.
+  temp.track();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 spaces but found 2.](https://app.codacy.com/manual/BetaWorks-NodeRED-Watson/node-red-node-watson/pullRequest?prid=4764899)